### PR TITLE
Cover 'order filtration' scenarios in API

### PR DESCRIPTION
**Pull request by TheMilek**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.13                   |
| Bug fix?        | no                                                     |
| New feature?    | yes                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no  |
| License         | MIT                                                          |

### CODE REVIEW

File: features/order/managing_orders/order_filtration/filtering_orders_by_products.feature [hunk 1]
```diff
@@ -17,15 +17,15 @@ Feature: Filtering orders by products
         And the customer chose "Free" shipping method to "United States" with "Offline" payment
         And I am logged in as an administrator
 
-    @ui @javascript
+    @ui @api @javascript
```
Comment (Wojdylak) [inline features/order/managing_orders/order_filtration/filtering_orders_by_products.feature:20]: Can we establish a convention? In the system, we have three variations: `@api @ui @javascript`, `@ui @api @javascript`, and `@ui @javascript @api`. I know it's not a problem of this PR.

File: src/Sylius/Behat/Context/Api/Admin/ManagingOrdersContext.php [hunk 2]
```diff
@@ -62,6 +65,113 @@ public function iBrowseOrders(): void
...
+     * @When I specify filter total being greater than :total
+     */
+    public function iSpecifyFilterTotalBeingGreaterThan(string $total): void
+    {
+        if (str_contains($total, '.')) {
+            $total = str_replace('.', '', $total);
+            $this->client->addFilter('total[gt]', $total);
+
+            return;
+        }
```
Comment (Wojdylak) [inline src/Sylius/Behat/Context/Api/Admin/ManagingOrdersContext.php:140]: I suggest creating a private method that either removes `.` or adds `00`. This method could be used here and method below.

File: src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml [hunk 3]
```diff
@@ -85,6 +85,55 @@
...
         </service>
 
+        <service id="sylius.api.order_date_filter" parent="api_platform.doctrine.orm.date_filter" public="true">
+            <argument type="collection">
+                <argument key="checkoutCompletedAt">exclude_null</argument>
+            </argument>
+            <tag name="api_platform.filter" />
+        </service>
+
+        <service id="sylius.api.order_product_filter" parent="api_platform.doctrine.orm.search_filter" public="true">
```
Comment (Wojdylak) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services/filters.xml:95]: It is only a suggestion to connect all `search_filter` for `order` in one filter.

**Review by Wojdylak — APPROVED** [review]

File: src/Sylius/Behat/Context/Api/Admin/ManagingOrdersContext.php [hunk 4]
```diff
@@ -62,6 +65,113 @@ public function iBrowseOrders(): void
...
+     */
+    public function iSpecifyFilterDateToAs(string $dateTime): void
+    {
+        $this->client->addFilter('checkoutCompletedAt[before]', $dateTime);
+    }
+
+
+    /**
+     * @When I filter by product :productName
+     * @When I filter by products :firstProduct and :secondProduct
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Admin/ManagingOrdersContext.php:103]:
```suggestion
     * @When I filter by products :firstProductName and :secondProductName
```

File: src/Sylius/Behat/Context/Api/Admin/ManagingOrdersContext.php [hunk 5]
```diff
@@ -62,6 +65,113 @@ public function iBrowseOrders(): void
...
+     */
+    public function iFilterByProduct(string ...$productNames): void
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Admin/ManagingOrdersContext.php:105]:
```suggestion
    public function iFilterByProduct(string ...$productsNames): void
```

File: src/Sylius/Behat/Context/Api/Admin/ManagingOrdersContext.php [hunk 6]
```diff
@@ -62,6 +65,113 @@ public function iBrowseOrders(): void
...
+     * @When I specify filter total being less than :total
+     */
+    public function iSpecifyFilterTotalBeingLessThan(string $total): void
+    {
+        $this->client->addFilter('total[lt]', $total. '00');
+    }
+
+    /**
+     * @When I filter by variant :variantName
+     * @When I filter by variants :firstVariant and :secondVariant
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Admin/ManagingOrdersContext.php:155]:
```suggestion
     * @When I filter by variants :firstVariantName and :secondVariantName
```

File: src/Sylius/Behat/Context/Api/Admin/ManagingOrdersContext.php [hunk 7]
```diff
@@ -344,6 +446,32 @@ public function iShouldSeeOrderWithNumber(string $orderNumber): void
         );
     }
 
+    /**
+     * @Then I should not see an order with :orderNumber number
+     */
+    public function iShouldNotSeeOrderWithNumber(string $orderNumber)
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Admin/ManagingOrdersContext.php:452]:
```suggestion
    public function iShouldNotSeeOrderWithNumber(string $orderNumber): void
```

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, Kamil! :tada:

**Pull request merged**